src/java/picard/sam/ValidateSamFile.java

```diff
@@ -58,20 +58,24 @@
 )
 public class ValidateSamFile extends CommandLineProgram {
     static final String USAGE_SUMMARY = "Validates a SAM or BAM file.  ";
-    static final String USAGE_DETAILS = "This tool reports on the validity of a SAM or BAM file relative to the SAM format specification " +
-            "(see http://samtools.github.io/hts-specs/SAMv1.pdf), which is useful for troubleshooting errors encountered with other tools " +
-            "that may be caused by improper formatting.<br /><br />" +
-            "By default, the tool runs in VERBOSE mode and will exit after finding 100 errors and output them to the " +
-            "console (stdout). It is often practical to start by running this tool with the SUMMARY mode option, which summarizes the " +
-            "\"errors\" and \"warnings\". Consequently, specific validation warnings or errors that are of lesser concern can be ignored " +
-            "using the IGNORE and/or IGNORE_WARNINGS arguments in order to focus on blocking errors. " +
-            "<br />" +
+    static final String USAGE_DETAILS = "<p>This tool reports on the validity of a SAM or BAM file relative to the SAM format specification.  " +
+            "For detailed information on the SAM specification, please see the SAM specification sheet <a href='http://samtools.github.io/hts-specs/SAMv1.pdf'>here</a>.</p> " +
+
+            "<p>Although the default setting for the tool runs in MODE=VERBOSE mode we highly recommend that you run the tool initially in MODE=SUMMARY.  " +
+            "Running the tool in MODE=SUMMARY, will output a high-level overview including a list and a count of all 'WARNING' and/or 'ERROR' messages for your SAM/BAM file.  " +
+            "Subsequently running the tool in MODE=VERBOSE will output a detailed list of every record with a 'WARNING' and/or 'ERROR' message."+
+            "  Validation errors or warnings that are of lesser concern can be ignored using the IGNORE and/or IGNORE_WARNINGS arguments " +
+            "to prioritize blocking errors.</p> " +
+            "<p>After identifying and fixing your 'warnings/errors', we recommend that you rerun this tool to validate your SAM/BAM file prior to" +
+            " inputting directly into <a href='http://broadinstitute.github.io/picard/'>Picard</a> or <a href='https://www.broadinstitute.org/gatk/'>GATK</a>.</p>" +
             "<h4>Usage example:</h4>" +
             "<pre>" +
             "java -jar picard.jar ValidateSamFile \\<br />" +
-            "     I=input.bam \\<br />" +
-            "     MODE=SUMMARY" +
+            "      I=input.bam \\<br />" +
+            "      MODE=SUMMARY" +
             "</pre>" +
+            "" +
+            "<p>To obtain a complete list with descriptions of both 'ERROR' and 'WARNING' messages, please visit the <a href='http://gatkforums.broadinstitute.org/gatk/discussion/6956'>Common Problems</a> forum link."+
```

Comment:
Sorry, I didn't see the appendix!

That said, it would be great if SAMValidationError.Type could be exposed like here:  http://static.javadoc.io/com.github.samtools/htsjdk/1.128/htsjdk/samtools/SAMValidationError.Type.html or similar, that would get updated when/if validation errors are added. 

Also, it seems that some documentation is missing from that class, could you add one-liners (e.g. MISMATCH_READ_LENGTH_AND_QUALS_LENGTH )? 

Thanks for doing this!!